azure/services/scalesetvms/scalesetvms.go

```diff
@@ -122,24 +237,53 @@ func (s *Service) Delete(ctx context.Context) error {
 		return nil
 	}
 
-	// since the future was nil, there is no ongoing activity; start deleting the instance
+	// since the future was nil, there is no ongoing activity; start deleting the vm
 	future, err := s.Client.DeleteAsync(ctx, resourceGroup, vmssName, instanceID)
 	if err != nil {
 		if azure.ResourceNotFound(err) {
 			// already deleted
 			return nil
 		}
-		return errors.Wrapf(err, "failed to delete instance %s/%s", vmssName, instanceID)
+		return errors.Wrapf(err, "failed to delete vm %s/%s", vmssName, instanceID)
 	}
 
 	s.Scope.SetLongRunningOperationState(future)
 
-	log.V(4).Info("checking if the instance is done deleting")
+	log.V(4).Info("checking if the vm is done deleting")
 	if _, err := s.Client.GetResultIfDone(ctx, future); err != nil {
-		// fetch instance to update status
+		// fetch vm to update status
 		return errors.Wrap(err, "failed to get result of long running operation")
 	}
 
 	s.Scope.DeleteLongRunningOperationState(instanceID, serviceName, infrav1.DeleteFuture)
 	return nil
 }
+
+// VMSSFlexVMGetter is an interface for getting all the required information to create/update/delete an Azure resource.
+type VMSSFlexVMGetter struct {
+	Name          string
+	ResourceGroup string
+}
+
+// ResourceName returns the name of the resource.
+func (vm *VMSSFlexVMGetter) ResourceName() string {
+	return vm.Name
+}
+
+// OwnerResourceName returns the name of the resource that owns the resource
+// in the case that the resource is an Azure subresource.
+func (vm *VMSSFlexVMGetter) OwnerResourceName() string {
+	return ""
+}
+
+// ResourceGroupName returns the name of the resource group the resource is in.
+func (vm *VMSSFlexVMGetter) ResourceGroupName() string {
+	return vm.ResourceGroup
+}
+
+// Parameters takes the existing resource and returns the desired parameters of the resource.
+// If the resource does not exist, or we do not care about existing parameters to update the resource, existing should be nil.
+// If no update is needed on the resource, Parameters should return nil.
+func (vm *VMSSFlexVMGetter) Parameters(existing interface{}) (params interface{}, err error) {
```

Comment:
what's going on with this no-op Parameters func here? Was this an attempt to implement the ResourceSpecGetter interface but without using it yet since we're not yet using the async package?